packages/notebook-extension/src/index.ts

```diff
@@ -2501,7 +2501,13 @@ function addCommands(
         );
       }
     },
-    isEnabled,
+    isEnabled: args => {
+      const current = getCurrent(tracker, shell, args);
+      if (!current) {
+        return false;
+      }
+      return current.content.activeCellIndex >= 1;
```

Comment:
As mentioned by Mike, the status `isEnabled` needs to be updated if the active cell changes to fix the bug seen when restoring the first cell for example.

You can achieve this by triggering `commands.notifyCommandChanged(<command id>)`.

Now the trouble is that this should be triggered for all notebooks opened. Luckily, this is already done for you if you listen to `tracker.activeCellChanged`.

So the final code will look like,

```ts
tracker.activeCellChanged.connect(() => {
  commands.notifyCommandChanged(CommandIDs.moveDown);
  commands.notifyCommandChanged(CommandIDs.moveUp);
})
```

Reply:
Thanks for the precisions @fcollonval. I tried to take into account the last comment in commit 
fdb0baa. In my local tests, the tentative fix did not solve the issue when removing first cell and restoring it.

Reply:
Good point @krassowski . 

The notebook signal documentation states:

https://github.com/jupyterlab/jupyterlab/blob/013b6ab817b929f5eaebaec7b282d0d0f52527c1/packages/notebook/src/widget.ts#L1237-L1243

So for me the current code is erroneous:

https://github.com/jupyterlab/jupyterlab/blob/013b6ab817b929f5eaebaec7b282d0d0f52527c1/packages/notebook/src/widget.ts#L1321-L1327

We should trigger `this._activeCellChanged.emit(cell);` if the new active index changes even if the widget is the same.

What do you think?

Reply:
@krassowski @krassowski Thanks for your feedbacks. I have tried to modify `widget.ts` to cover the case "the new active index changes" in order to fit with the explaining notes relative to the `activeCellChanged`. This seems to work on the previous simple 2-cell test suggested.

Reply:
@fcollonval the way I read the documentation the current implementation is not erroneous. I think that the second part "or cell at the active index changing" refers to a case when say a code cell is converted to e.g. markdown cell (so the cell at the active index _changes_).

It feels to me like `activeCellChanged` signal is doing what I would expect it to do. We could have another signal, say `cellPositionChanged` to deal with situations like this one.

@HaudinFlorence thanks for verifying that this would fix it. We may want to think a little more on what is the clean way to approach it (while I am proposing adding a new signal it is just brainstorming - we are usually reluctant to extend API without taking more time to think about it as it is difficult to change later).

What do you both think about using `notebook.model.cells.changed` signal instead?

Reply:
@krassowski Thanks for your feedback. I can try to use `notebook.model.cells.changed`. I will wait for @fcollonval's opinion before trying.

Reply:
@krassowski I was actually referring to the first clause _the active index changing_ :wink:. The problematic case highlights a problem with that clause as restoring a delete cell is not changing the active cell but is changing the active index. No emitting the signal is such a case feels wrong.

Pinging other devs for input on that signal problem: @jasongrout @afshin @blink1073 @jtpio 

Another edge case (not related to this PR), prior to dropping modelDB, moving a cell was not triggering that signal (as the widget was not changed). But now it does as a move is add + delete. But when the move feature lands in Yjs/Ypy, we will fallback to the previous behavior. So I would rather emitted more often but ensure better consistent (more independent of the implementation).

Reply:
I am happy to move forward here (after making sure we don't emit spurious update as in your suggestion).